### PR TITLE
slo: add CI stats from sippy

### PR DESCRIPTION
**Pull request by sttts**
~~Requires https://github.com/openshift/sippy/pull/107 (using fake bump currently).~~ merged.

Default obligation is 100 ci-fail-rate now. Reasonable default will be set in shiftzilla_cfg.yaml.

### CODE REVIEW

File: pkg/slo/slo.go [hunk 1]
```diff
@@ -114,7 +184,31 @@ func getPMScoreResult(bugMap bugs.TeamMap, sloData sloAPI.Data, teamInfo teams.T
...
+		if !found {
+			continue
+		}
+
+		if pr, ok := passRate.PassRates["latest"]; ok && pr.Percentage < minPass {
+			minPass = pr.Percentage
+		}
+	}
+	obligation := int(sloData.Count)
+	if obligation == 0 {
```
Comment (eparis) [inline pkg/slo/slo.go]: We should do defaulting in the shiftzilla config. You should be able to test that locally creating an overwrite yaml file and use --overwrite-team-data on the team exporter.

Comment (sttts) [reply pkg/slo/slo.go]: Fixed.

So this means we need a shiftzilla PR first?

Comment (sttts) [reply pkg/slo/slo.go]: https://github.com/openshift/li/pull/3994

File: pkg/slo/slo.go [hunk 2]
```diff
@@ -67,6 +71,72 @@ func GetBugMaps(bugData *bugs.BugData) map[string]bugs.TeamMap {
...
+
+	result := map[string]sippyv1.MinimumPassRatesByComponent{}
+	for _, c := range r[version].MinimumJobPassRatesByComponent {
+		result[c.Name] = c
+	}
+	return result, nil
+}
+
+// currentVersion returns the lowest x.y version that has a x.y.0 target.
+func CurrentVersion(releases map[string]teams.ReleaseInfo) (string, error) {
```
Comment (mfojtik) [inline pkg/slo/slo.go:104]: nit: case

File: pkg/slo/slo.go (hunk 1 above)
Comment (derekwaynecarr) [reply pkg/slo/slo.go]: i merged https://github.com/openshift/li/pull/3994

File: pkg/slo/slo.go [hunk 3]
```diff
@@ -67,6 +71,72 @@ func GetBugMaps(bugData *bugs.BugData) map[string]bugs.TeamMap {
...
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to download release-%s sippy stats (%d): %v", version, resp.StatusCode, string(body))
+	}
+
+	type Report map[string]struct {
+		MinimumJobPassRatesByComponent []sippyv1.MinimumPassRatesByComponent `json:"minimumJobPassRatesByComponent"`
+	}
+	var r Report
+	if err := json.Unmarshal(body, &r); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal release-%s suppy state: %v", version, err)
```
Comment (derekwaynecarr) [inline pkg/slo/slo.go]: nit: sippy

File: pkg/slo/slo.go (hunk 1 above)
Comment (sttts) [reply pkg/slo/slo.go]: merged

File: pkg/slo/slo.go (hunk 2 above)
Comment (sttts) [reply pkg/slo/slo.go:104]: fixed

File: pkg/slo/slo.go (hunk 3 above)
Comment (sttts) [reply pkg/slo/slo.go]: fixed

**Comment by derekwaynecarr** [comment]
changes look good.

merging while @eparis is out.

/approve
/lgtm

**Pull request merged**